apps/client/src/components/Pokedex/Pokedex.tsx

```diff
@@ -24,6 +24,13 @@ const Pokedex: React.FC<PokedexProps> = ({ basicPokemons = [] }) => {
 
   useOnReachedBottom(divRef, loadMorePokemons);
 
+  // To Re-render the component when the basicPokemons change
+  const [prevShowingPokemons, setPrevShowingPokemons] = useState(basicPokemons);
```

Comment:
Don't be cheap on couple letters lol

```suggestion
  const [previousShowingPokemons, setPreviousShowingPokemons] = useState(basicPokemons);
```